src/scala/microsoft-spark-2-4/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -123,6 +125,9 @@ object DotnetRunner extends Logging {
       if (!runInDebugMode) {
         var returnCode = -1
         var process: Process = null
+        var enableLogRedirection: Boolean = false
+        var stderrBuffer: CircularBuffer = null
```

Comment:
Let's set these once as `val`s and change to `stderrBuffer: Option[CircularBuffer]`